csharp/ql/src/semmle/code/csharp/ir/rangeanalysis/RangeAnalysis.qll

```diff
@@ -233,19 +233,25 @@ private class SafeCastInstruction extends ConvertInstruction {
  * Holds if `typ` is a small integral type with the given lower and upper bounds.
  */
 private predicate typeBound(IntegralType typ, int lowerbound, int upperbound) {
-  typ instanceof SignedIntegralType and typ.getSize() = 1 and lowerbound = -128 and upperbound = 127
+  typ instanceof SignedIntegralType and
+  typ.getSize() = 1 and
+  lowerbound = typ.minValue() and
```

Comment:
Since `minValue` and `maxValue` are on `IntegralType`, they could be moved outside thus
```
lowerbound = type.minValue() and 
upperbound = ... and
(
  ...
)
```